x-pack/plugins/uptime/server/lib/alerts/uptime_alert_wrapper.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ILegacyScopedClusterClient } from 'kibana/server';
+import { AlertExecutorOptions, AlertType, State } from '../../../../alerts/server';
+import { savedObjectsAdapter } from '../saved_objects';
+import { DynamicSettings } from '../../../common/runtime_types';
+
+export interface UptimeAlertType extends Omit<AlertType, 'executor' | 'producer'> {
+  executor: (
+    options: AlertExecutorOptions,
+    callES: ILegacyScopedClusterClient['callAsCurrentUser'],
+    dynamicSettings: DynamicSettings
+  ) => Promise<State | void>;
+}
+
+export const uptimeAlertWrapper = (uptimeAlert: UptimeAlertType) => ({
```

Comment:
This is an interesting pattern, but why not use classes and inheritance instead of the juggling of Omit and various method definitions? While this is functionally equivalent, it's more complex and less semantically clear IMHO.

It feels a bit like traditional OO is avoided in JS/TS, but this seems like a good situation to use it.

Reply:
it's being used in other places in uptime server, that's why i stick with this. let me see if i can spend some time to improve it.

Reply:
@andrewvc would you mind putting off this discussion for later? we do have precedence to use the same pattern 
https://github.com/elastic/kibana/blob/master/x-pack/plugins/uptime/server/rest_api/uptime_route_wrapper.ts 

it's easy to write i think, writing a class will require defining all the properties which will be part of the return object, setting, writing a wrapper function etc. to me it seems like more effort.